RedfishPkg/Include/Protocol/EdkIIRedfishCredential2.h

```diff
@@ -0,0 +1,123 @@
+  @retval EFI_ACCESS_DENIED        SecureBoot is disabled after EndOfDxe.
+  @retval EFI_INVALID_PARAMETER    This or AuthMethod or UserId or Password is NULL.
+  @retval EFI_OUT_OF_RESOURCES     There are not enough memory resources.
+  @retval EFI_UNSUPPORTED          Unsupported authentication method is found.
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EDKII_REDFISH_CREDENTIAL2_PROTOCOL_GET_AUTH_INFO)(
+  IN  EDKII_REDFISH_CREDENTIAL2_PROTOCOL    *This,
+  OUT EDKII_REDFISH_AUTH_METHOD             *AuthMethod,
+  OUT CHAR8                                 **UserId,
+  OUT CHAR8                                 **Password
+  );
+
+/**
+  Notify the Redfish service provide to stop provide configuration service to this platform.
+
+  This function should be called when the platform is about to leave the safe environment.
+  It will notify the Redfish service provider to abort all login sessions, and prohibit
+  further login with original auth info. GetAuthInfo() will return EFI_UNSUPPORTED once this
+  function is returned.
+
+  @param[in]   This                Pointer to EDKII_REDFISH_CREDENTIAL2_PROTOCOL instance.
+  @param[in]   ServiceStopType     Reason of stopping Redfish service.
+
+  @retval EFI_SUCCESS              Service has been stopped successfully.
+  @retval EFI_INVALID_PARAMETER    This is NULL.
+  @retval Others                   Some error happened.
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EDKII_REDFISH_CREDENTIAL2_PROTOCOL_STOP_SERVICE)(
+  IN  EDKII_REDFISH_CREDENTIAL2_PROTOCOL            *This,
+  IN  EDKII_REDFISH_CREDENTIAL_STOP_SERVICE_TYPE    ServiceStopType
+  );
+
+/**
+  Register Redfish service instance so protocol knows that some module uses bootstrap account .
+
+  @param[in]  This            Pointer to EDKII_REDFISH_CREDENTIAL2_PROTOCOL instance.
+  @param[in]  RedfishService  Redfish service instance to register.
+
+  @retval EFI_SUCCESS         This Redfish service instance has been registered successfully.
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EDKII_REDFISH_CREDENTIAL2_PROTOCOL_REGISTER_REDFISH_SERVICE)(
+  IN  EDKII_REDFISH_CREDENTIAL2_PROTOCOL    *This,
+  IN  REDFISH_SERVICE                       RedfishService
+  );
+
+/**
+  Unregister Redfish service instance and delete the bootstrap account
+  when all registered services unregistered.
+
+  @param[in]  This            Pointer to EDKII_REDFISH_CREDENTIAL2_PROTOCOL instance.
+  @param[in]  RedfishService  Redfish service instance to unregister.
+
+  @retval EFI_SUCCESS       This Redfish service instance has been unregistered successfully.
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EDKII_REDFISH_CREDENTIAL2_PROTOCOL_UNREGISTER_REDFISH_SERVICE)(
+  IN  EDKII_REDFISH_CREDENTIAL2_PROTOCOL    *This,
+  IN  REDFISH_SERVICE                       RedfishService
+  );
+
+struct _EDKII_REDFISH_CREDENTIAL2_PROTOCOL {
+  EDKII_REDFISH_CREDENTIAL2_PROTOCOL_GET_AUTH_INFO                 GetAuthInfo;

```

Comment:
HI @igorkulchytskyy, let's add a version control filed in the protocol structure. So we don't have to introduce a new protocol again when adding a new interface in the future.

Reply:
Hi Abner, I agree with you. I will add a version control. Thank you for comment.

Reply:
Hi Abner,
I added Revision of the protocol.